cobalt/watchdog/watchdog.cc

```diff
@@ -497,11 +550,71 @@ bool Watchdog::Unregister(const std::string& name, bool lock) {
   return result;
 }
 
+std::shared_ptr<Client> Watchdog::UnregisterByClient(
+    std::shared_ptr<Client> client) {
+  if (is_disabled_) return nullptr;
+
+  starboard::ScopedLock scoped_lock(mutex_);
+
+  std::string name = "";
+  if (client) name = client->name;
+
+  // Unregisters.
+  for (auto it = client_list_.begin(); it != client_list_.end(); it++) {
+    if (client == *it) {
+      client_list_.erase(it);
+      SB_DLOG(INFO) << "[Watchdog] Unregistered: " << name;
+      return nullptr;
+    }
+  }
+  return client;
```

Comment:
Why does this function return a client? What's it used for?

Reply:
I wanted to have it symmetric with register and return the original client pointer, but thinking about it I think that's overly complex. Will revert to using bools.